tests/core/scenario/test_scenario_manager.py

```diff
@@ -1553,3 +1553,35 @@ def test_filter_scenarios_by_creation_datetime():
     )
     assert len(filtered_scenarios) == 1
     assert [s_1_1] == filtered_scenarios
+
+
+def test_clone_scenario():
+    dn_config_1 = Config.configure_pickle_data_node("dn_1", scope=Scope.SCENARIO)
+    dn_config_2 = Config.configure_pickle_data_node("dn_2", scope=Scope.SCENARIO)
+    additional_dn_config_1 = Config.configure_data_node("additional_dn_1", scope=Scope.SCENARIO)
+    task_config_1 = Config.configure_task("task_1", print, [dn_config_1], [dn_config_2])
+    scenario_config_1 = Config.configure_scenario("scenario_1", [task_config_1], [additional_dn_config_1])
+    scenario = _ScenarioManager._create(scenario_config_1)
+    old_scenario_id = scenario.id
+
+    assert len(_ScenarioManager._get_all()) == 1
+    assert len(_DataManager._get_all()) == 3
+    assert len(_TaskManager._get_all()) == 1
+
+    new_scenario = _ScenarioManager._clone(scenario)
+    old_scenario = _ScenarioManager._get(old_scenario_id)
+
+    assert old_scenario.id != new_scenario.id
```

Comment:
We don't want to change the scenario (even in memory). we want to create a new one.

THe following must be tested.

```suggestion
    assert old_scenario.id != new_scenario.id
    assert old_scenario.id == scenario.id
```

Unfortunately, this means we need to change the implementation.

Reply:
fixed! :D